x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/step_select_agent_policy.tsx

```diff
@@ -92,6 +72,12 @@ export const StepSelectAgentPolicy: React.FunctionComponent<{
     }, {});
   }, [agentPolicies]);
 
+  // Selected agent policy state
+  const [selectedPolicyId, setSelectedPolicyId] = useState<string | undefined>(
+    agentPolicy?.id ??
+      (selectedAgentPolicyId || (agentPolicies.length === 1 ? agentPolicies[0].id : undefined))
```

Comment:
need to check|test but I think here`agentPolicies` will always be empty during initialization

Reply:
So we agreed on the UX design that the Agent policy will be selected only if there is one of them, otherwise it won't be selected by default. This should be working, it is covered by cypress test too.

Reply:
Here I have agentPolicies and none are selected, I think the useState is initialized with undefined and it will not update the value when agentPolicies are retrieved.